compiler/plugins/target/ROCM/ROCMTarget.cpp

```diff
@@ -76,6 +81,34 @@ struct ROCmOptions {
                  "specific unprefixed microkernels to enable, e.g. `mmt4d`."));
   }
 
+  LogicalResult verify(mlir::Builder &builder) const {
+    if (GPU::normalizeHIPTarget(targetChip).empty()) {
+      return emitError(builder.getUnknownLoc(), "Unknown ROCm target '")
+             << targetChip << "'";
+    }
+    SmallVector<StringRef> features;
+    llvm::SplitString(targetFeatures, features, ",");
+    for (StringRef f : features) {
+      if (f.ends_with("+") || f.ends_with("-") ||
+          !(f.starts_with("+") || f.starts_with("-"))) {
+        return emitError(builder.getUnknownLoc(),
+                         "ROCm target feature must be prefixed with '+' or "
+                         "'-'; but seen '")
+               << f << "'";
+      }
+      StringRef feature = f.substr(1);
+      if (feature != "sramecc" && feature != "xnack") {
+        // We only support these two features to be set explicitly. Features
+        // like wavefrontsize is controlled and tuned by the compiler.
+        return emitError(builder.getUnknownLoc(),
+                         "ROCm target feature can only be 'sramecc' or "
+                         "'xnack'; but seen '")
+               << feature << "'";
```

Comment:
Wouldn't it be better to make this a black list of things controlled by the compiler instead of an allow list? This looks like good cleanup regardless so not blocking.

Reply:
The AMDGPU backend in LLVM pretty much only accepts these two for describing the targets. Read the few functions starting at https://github.com/llvm/llvm-project/blob/d86b68a/llvm/lib/Target/AMDGPU/Utils/AMDGPUBaseInfo.cpp#L745. So I think it's okay to be explicit here.

Reply:
I see, makes sense then